include/ddc/kernels/splines/bsplines_non_uniform.hpp

```diff
@@ -28,39 +28,49 @@ struct NonUniformBSplinesBase
 
 } // namespace detail
 
-/// NonUniformPointSampling specialization of BSplines
+/**
+ * The type of a non-uniform BSplines 1D basis.
+ *
+ * @tparam Tag The tag identifying the continuous dimension which supports the building of the BSplines.
+ * @tparam D The degree of the BSplines.
+ */
 template <class Tag, std::size_t D>
 class NonUniformBSplines : detail::NonUniformBSplinesBase
 {
     static_assert(D > 0, "Parameter `D` must be positive");
 
 public:
+    /// @brief The tag identifying the continuous dimension which supports the building of the BSplines.
     using tag_type = Tag;
 
-
+    /// @brief The discrete dimension identifying BSplines.
     using discrete_dimension_type = NonUniformBSplines;
 
-public:
+    /// @brief The degree of BSplines.
     static constexpr std::size_t degree() noexcept
     {
         return D;
     }
 
+    /// @brief Indicates if the BSplines are periodic or not.
     static constexpr bool is_periodic() noexcept
     {
         return Tag::PERIODIC;
     }
 
+    /// @brief Indicates if the BSplines are radial or not (should be deprecated soon because this concept is not in the scope of DDC).
     static constexpr bool is_radial() noexcept
     {
         return false;
     }
 
+    /// @brief Indicates if the BSplines are uniform or not (this is not the case here).
     static constexpr bool is_uniform() noexcept
```

Comment:
Somewhere in this file, you should explain what is meant by "uniform" this is often misunderstood and different words are sometimes used for this context. I suspect adding a sentence here and elaborating in the class docstring

Reply:
`whose knots are uniformly distributed` is correct ?